### PR TITLE
Add letter frequency functionality

### LINKED ISSUE #418: Lab 03 - Letter Frequency
# Lab 3 - Letter frequency

- Create an executable go program in directory `03_letters/USERNAME`
- Write a function that returns a mapping of each letter to its frequency:

```
func letters(s string) map[rune]int
```

Write a function that returns a sorted slice of strings with elements `"{key}:{val}"`. Use package [sort](https://golang.org/pkg/sort/):

```
func sortLetters(m map[rune]int) []string
```

Call `fmt.Println(strings.Join(sortLetters(letters("aba"))), "\n")` in `main` to print:

```
a:2
b:1
```

Bonus points: comprehensive tests

### PR DESCRIPTION
**Pull request by patrickmarabeas**
Implements two functions. One which creates a frequency map of given runes, the other sorts the map and returns a slice of "k:v" strings

Fixes #418 

Review of colleague's PR #527

### CODE REVIEW

File: 03_letters/patrickmarabeas/main.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	// Increment key's value by one for each instance
+	// Starts with the zero value of the ValueType (int)
+	for _, v := range s {
+		result[v]++
+	}
+
+	return result
+}
+
+// Sorted letter kvp slice
```
Comment (runnerdave) [inline 03_letters/patrickmarabeas/main.go]: what is kvp?

File: 03_letters/patrickmarabeas/main.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	result := make([]string, length)
+	// Create a slice of runes from map keys
+	runes := make([]rune, length)
+	i := 0
+	for k := range m {
+		runes[i] = k
+		i++
+	}
+	// Sort the slice of runes
+	// The less function that sort implements takes int32 which aligns with rune
```
Comment (runnerdave) [inline 03_letters/patrickmarabeas/main.go:38]: I believe these comments are unnecessary, the code should be self explanatory

File: 03_letters/patrickmarabeas/main_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		}
+	})
+}
+
+func TestLetters(t *testing.T) {
+	var cases = map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"Letters": {
```
Comment (runnerdave) [inline 03_letters/patrickmarabeas/main_test.go:30]: you may consider testing more complex combinations such as uppercases and longer strings, as well as empty strings

**Review by runnerdave — CHANGES_REQUESTED** [review]
minor changes, but the main important is your PR description, I believe you should review other person's PR and reference it there, as well as listing your changes in the description

File: 03_letters/patrickmarabeas/main.go (hunk 1 above)
Comment (patrickmarabeas) [reply 03_letters/patrickmarabeas/main.go]: key value pair

File: 03_letters/patrickmarabeas/main.go (hunk 2 above)
Comment (patrickmarabeas) [reply 03_letters/patrickmarabeas/main.go:38]: As a Go scrub, it really isn't. Initially I was using casting to solve this - but that felt wrong. This documents *for myself* how I got around the need to do so, and why this works (how the types align).

File: 03_letters/patrickmarabeas/main_test.go (hunk 3 above)
Comment (patrickmarabeas) [reply 03_letters/patrickmarabeas/main_test.go:30]: Some additional tests added.

**Review by runnerdave — CHANGES_REQUESTED** [review]
Hi, saw you addressed the extra test, but you also need to improve your PR description

**Comment by patrickmarabeas** [comment]
PR has been updated with reviewed #

**Comment by patrickmarabeas** [comment]
Updated doc string

**Review by pentaphobe — APPROVED** [review]
LGTM!

**Review by runnerdave — APPROVED** [review]
👍

**Pull request merged**